dotcom-rendering/cdk/lib/renderingStack.ts

```diff
@@ -0,0 +1,128 @@
+import { type Alarms, GuEc2App } from '@guardian/cdk';
+import { AccessScope } from '@guardian/cdk/lib/constants';
+import type { NoMonitoring } from '@guardian/cdk/lib/constructs/cloudwatch';
+import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
+import {
+	GuStack as CDKStack,
+	GuDistributionBucketParameter,
+} from '@guardian/cdk/lib/constructs/core';
+import { GuCname } from '@guardian/cdk/lib/constructs/dns/dns-records';
+import { GuAllowPolicy } from '@guardian/cdk/lib/constructs/iam';
+import type { GuAsgCapacity } from '@guardian/cdk/lib/types';
+import { type App as CDKApp, Duration } from 'aws-cdk-lib';
+import type { InstanceSize } from 'aws-cdk-lib/aws-ec2';
+import { InstanceClass, InstanceType, Peer } from 'aws-cdk-lib/aws-ec2';
+import { StringParameter } from 'aws-cdk-lib/aws-ssm';
+import { getUserData } from './userData';
+
+export interface RenderingCDKStackProps extends Omit<GuStackProps, 'stack'> {
+	guApp: `${'article' | 'facia' | 'misc' | 'interactive'}-rendering` | string; // TODO - remove string
+	domainName: string;
+	instanceSize: InstanceSize;
+	scaling: GuAsgCapacity;
+}
+
+/** DCR infrastructure provisioning via CDK */
+export class RenderingCDKStack extends CDKStack {
+	constructor(scope: CDKApp, id: string, props: RenderingCDKStackProps) {
+		super(scope, id, {
+			...props,
+			// Any version of this app should run in the eu-west-1 region
+			env: { region: 'eu-west-1' },
+			// Set the stack within the constructor as this won't vary between apps
+			stack: 'frontend',
+		});
+
+		const { stack: guStack, region, account } = this;
+		const { guApp, stage, instanceSize, scaling, domainName } = props;
+
+		const artifactsBucket =
+			GuDistributionBucketParameter.getInstance(this).valueAsString;
+
+		const monitoringConfiguration =
+			stage === 'PROD'
+				? ({
+						snsTopicName: `Frontend-${stage}-CriticalAlerts`,
+						// TODO – how does this overlap with the DevX debug dashboard?
+						http5xxAlarm: {
```

Comment:
How did we come up with the values for `tolerated5xxPercentage` and `numberOfMinutesAboveThresholdBeforeAlarm`?